druid/src/main/java/org/apache/calcite/adapter/druid/DruidJsonFilter.java

```diff
@@ -294,11 +291,11 @@ abstract class DruidJsonFilter implements DruidJson {
     if (columnName == null) {
       return null;
     }
-    if (e.getKind() != SqlKind.NOT_IN) {
+    if (e.getKind() == SqlKind.DRUID_IN) {
       return new DruidJsonFilter.JsonInFilter(columnName, listBuilder.build(), extractionFunction);
     } else {
-      return toNotDruidFilter(
-          new DruidJsonFilter.JsonInFilter(columnName, listBuilder.build(), extractionFunction));
+      return new DruidJsonFilter.JsonNotInFilter(columnName,
```

Comment:
I am not too familiar with Druid, but you haven't provided enough evidence the original behaviour is incorrect, and that your proposed changes are the correct ones.

`DruidAdapterIT` and `DruidAdapter2IT` seem to be the right places to add such a test.

Reply:
I had remove JsonNotInFilter class, and as you suggesstion I thought not need this new class.

Reply:
We need at least one test to show precisely the issue. The test should not pass before your change, and it should pass after it.

In this way we highlight what the change does, what problem it was there, and we protect the change against future possible regressions.